### PR TITLE
Refactoring code

### PR DESCRIPTION
**Pull request by meltsufin**
Replaces the Actions abstraction with service interfaces and removes the AppExecutor abstraction.

### CODE REVIEW

**Comment by meltsufin** [comment]
@patflynn, @loosebazooka PTAL

File: src/main/java/com/google/cloud/tools/app/impl/config/DefaultStageFlexibleConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+package com.google.cloud.tools.app.impl.config;
+
+import com.google.cloud.tools.app.api.deploy.StageFlexibleConfiguration;
+
+import java.io.File;
+
+/**
+ * Plain Java bean implementation of {@link StageFlexibleConfiguration}
+ */
+public class DefaultStageFlexibleConfiguration implements StageFlexibleConfiguration {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/config/DefaultStageFlexibleConfiguration.java]: why not just use the builders we had before?

Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/config/DefaultStageFlexibleConfiguration.java]: Just that this is simpler to generate. What's the advantage of the builders? To make these immutable?

File: src/main/java/com/google/cloud/tools/app/api/genconfig/GenConfigParams.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
+/**
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/api/genconfig/GenConfigParams.java]: are we getting rid of genconfig?

**Comment by patflynn** [comment]
I'm never going to be able to review this in a timely manner. Merge it and I'll do a post merge review.

**Pull request merged**